### PR TITLE
Fixes #26258: load organization after save.

### PR DESCRIPTION
**Pull request by waldenraines**
Also update an issue where the CDN URL was preventing changes
to the URL.

https://projects.theforeman.org/issues/26258

### CODE REVIEW

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 1]
```diff
@@ -60,7 +60,9 @@ class ManageManifestModal extends Component {
   };
 
   saveOrganization = () => {
-    this.props.saveOrganization({ redhat_repository_url: this.state.redhat_repository_url });
+    // TODO: Necessary because of https://projects.theforeman.org/issues/26420
+    this.props.saveOrganization({ redhat_repository_url: this.state.redhat_repository_url })
+      .then(this.props.loadOrganization);
```
Comment (waldenraines) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: @amirfefer @sharvit your thoughts on this approach? I don't know what else to do in this case.

**Comment by waldenraines** [comment]
[test katello]

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 1 above)
Comment (sharvit) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Can't we fix the server so it will return a valid response after put?

imo calling `loadOrganization` after `saveOrganization` is fine.
should it be done in the action `saveOrganization` itself since it produces a wrong state?

Comment (waldenraines) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: We should fix the API, I filed a bug about it but I don't have enough expertise to fix it.   I will try to move this to the `saveOrganization` action, good point :+1:

**Comment by waldenraines** [comment]
@sharvit updated.

**Comment by waldenraines** [comment]
[test katello]

**Comment by waldenraines** [comment]
[test katello]

**Comment by waldenraines** [comment]
[test katello]

**Comment by waldenraines** [comment]
[test katello]

File: webpack/scenes/Organizations/OrganizationActions.js [hunk 2]
```diff
@@ -49,6 +49,9 @@ export const saveOrganization = (extendedParams = {}) => (dispatch) => {
         type: SAVE_ORGANIZATION_SUCCESS,
         response: data,
       });
+
+      // TODO: Necessary because of https://projects.theforeman.org/issues/26420
+      dispatch(loadOrganization());
```
Comment (amirfefer) [inline webpack/scenes/Organizations/OrganizationActions.js]: this would execute two API requests (PUT and GET) when saving organization. 
wouldn't it be more reasonable fixing the API itself, and get the requested data in the first API ?

Comment (waldenraines) [reply webpack/scenes/Organizations/OrganizationActions.js]: Yes, it would.  However, this is for an upcoming release and I don't have the expertise (or time) to get this into foreman.

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js [hunk 3]
```diff
@@ -169,8 +169,8 @@ class ManageManifestModal extends Component {
                     <FormControl
                       id="cdnUrl"
                       type="text"
-                      value={this.state.redhat_repository_url || organization.redhat_repository_url || ''}
-                      onChange={this.updateRepositoryUrl}
+                      defaultValue={this.state.redhat_repository_url || organization.redhat_repository_url || ''}
+                      onBlur={this.updateRepositoryUrl}
```
Comment (johnpmitsch) [inline webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: why the change to `onBlur`? It seems to be working fine, just curious though

File: webpack/scenes/Organizations/OrganizationActions.js (hunk 2 above)
Comment (johnpmitsch) [inline webpack/scenes/Organizations/OrganizationActions.js]: do you need to run this with `dispatch`? It seems like `loadOrganization`, doesn't return an object to dispatch, but rather dispatches actions itself. Wouldn't just `loadOrganization()` work here?

**Comment by johnpmitsch** [comment]
Functionality is working fine :+1:  I just have two questions that I left as comments

File: webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js (hunk 3 above)
Comment (waldenraines) [reply webpack/scenes/Subscriptions/Manifest/ManageManifestModal.js]: Because it was updating on every change to the field and that was causing a weird issue where you couldn't update the field because of the delay in the update.

File: webpack/scenes/Organizations/OrganizationActions.js (hunk 2 above)
Comment (sharvit) [reply webpack/scenes/Organizations/OrganizationActions.js]: I don't think it will work since the `loadOrganization` needs the `dispatch` method.
`dispatch(loadOrganization())` is like doing `loadOrganization()(dispatch)`

Comment (johnpmitsch) [reply webpack/scenes/Organizations/OrganizationActions.js]: ah, ya I see that now, thanks for explaining!
